### ISSUE TITLE
TIFF image thumbnail broke using bimg with vips 8.10.1

### ISSUE THREAD

**Issue by vansante** [libvips/libvips#1815]
**Describe the bug**
Making a thumbnail of an TIFF image using `h2non/bimg` returns an error since version 8.10.x
 
**To Reproduce**

- Try to convert a tiff image using bimg. Use this for instance:

[testtiff.zip](https://github.com/libvips/libvips/files/5217153/testtiff.zip)

- You will get the error `tiff2vips: TIFF does not contain page 1`

**Expected behavior**
I expected a working thumbnail like in 8.9.x

**Actual behavior**
- You will get the error `tiff2vips: TIFF does not contain page 1`

**Environment**
 - OS: Debian Buster
 - Vips: 8.10.1 (manually compiled)

**Additional context**
- Curiously enough it seems to work using `vipsthumbnail`, so it might be an issue in bimg, although the error seems to suggest otherwise. Let me know if I should file a bugreport there though ;)

**Comment by jcupitt** [comment]
Hi @vansante,

I tried in master and I see:

```
$ vipsheader -a mri.tif 
mri.tif: 256x256 uchar, 1 band, b-w, tiffload
width: 256
height: 256
bands: 1
format: uchar
coding: none
interpretation: b-w
xoffset: 0
yoffset: 0
xres: 47.2441
yres: 47.2441
filename: mri.tif
vips-loader: tiffload
n-pages: 1
image-description: MatLab PGMWRITE file, saved 27-Aug-96
resolution-unit: in
orientation: 1
```

In 8.9 I see:

```
$ vipsheader -a mri.tif 
mri.tif: 256x256 uchar, 1 band, b-w, tiffload
width: 256
height: 256
bands: 1
format: uchar
coding: none
interpretation: b-w
xoffset: 0
yoffset: 0
xres: 47.2441
yres: 47.2441
filename: mri.tif
vips-loader: tiffload
n-pages: 1
image-description: MatLab PGMWRITE file, saved 27-Aug-96
resolution-unit: in
orientation: 1
```

So I think everything is the same, or at least I don't see a difference. Sorry, I think you'll need to ask bimg.

**Comment by jcupitt** [comment]
The error `tiff2vips: TIFF does not contain page 1` appears if you ask for a page that does not exist. For example:

```
john@yingna ~/pics $ vips copy mri.tif[page=0] x.png
john@yingna ~/pics $ vips copy mri.tif[page=1] x.png
tiff2vips: TIFF does not contain page 1
```

So your `mri.tif` only contains a single page (page zero). Something is asking for page 1, perhaps bimg.

**Comment by vansante** [comment]
Thanks for your quick reply, Ill do some further research and create an issue at bimg if it turns out its an issue there.

**Comment by jcupitt** [comment]
Oh yes, I think I see it too with this test prog:

```python
#!/usr/bin/python3

import sys
import pyvips

buf = open(sys.argv[1], "rb").read()

im = pyvips.Image.thumbnail_buffer(buf, 128)
buf_out = im.write_to_buffer(".jpg")

open(sys.argv[2], "wb").write(buf_out)
```

I see:

```
john@yingna ~/try $ ./try319.py ~/pics/mri.tif 
seen error
Traceback (most recent call last):
  File "./try319.py", line 8, in <module>
    im = pyvips.Image.thumbnail_buffer(buf, 128)
  File "/home/john/.local/lib/python3.8/site-packages/pyvips/vimage.py", line 135, in call_function
    return pyvips.Operation.call(name, *args, **kwargs)
  File "/home/john/.local/lib/python3.8/site-packages/pyvips/voperation.py", line 282, in call
    raise Error('unable to call {0}'.format(operation_name))
pyvips.error.Error: unable to call thumbnail_buffer
  tiff2vips: TIFF does not contain page 1
```

I'll investigate.

**Comment by jcupitt** [comment]
I think I found it -- a patch had been dropped. It's working now with 8.10 HEAD and master. I added a test too.

Thanks for the report! This fix will be in 8.10.2.

**Comment by vansante** [comment]
Awesome, thank you!
